File: packages/stripe_terminal_android/android/src/main/kotlin/com/facebook/react/bridge/UiThreadUtil.java
package com.facebook.react.bridge;

import org.jetbrains.annotations.NotNull;

import kotlin.Unit;
import kotlin.jvm.functions.Function0;

public class UiThreadUtil {
    public static void runOnUiThread(@NotNull Function0<Unit> function) {
        function.invoke();
    }
}

File: packages/stripe_android/android/src/main/kotlin/com/facebook/react/bridge/UiThreadUtil.kt
package com.facebook.react.bridge

import android.os.Handler
import android.os.Looper

class UiThreadUtil {
    companion object {
        private val handler = Handler(Looper.getMainLooper())

        fun runOnUiThread(function: () -> Unit) {
            handler.post {
                function()
            }
        }

        fun assertOnUiThread() {
            check(Looper.myLooper() == Looper.getMainLooper()) { "Expected to be called on the UI thread" }
        }
    }

}
